io/module_wrt_grid_comp.F90

```diff
@@ -2072,7 +2133,32 @@ subroutine wrt_run(wrt_comp, imp_state_write, exp_state_write,clock,rc)
              endif
           endif
 
-          filename = trim(wrt_int_state%wrtFB_names(nbdl))//'f'//trim(cfhour)//'.nc'
+          if (is_restart_bundle) then
+            write(time_restart,'(I4,I2.2,I2.2,".",I2.2,I2.2,I2.2)') cdate(1:6)
+
+            ! strip leading 'restart_' from a bundle name and replace it with a directory name 'RESTART/' to create actual file name
+            filename = 'RESTART/'//trim(time_restart)//'.'//trim(wrt_int_state%wrtFB_names(nbdl)(9:))//'.nc'
+
+            ! I hate this kind of inconsistencies
```

Comment:
This inconsistency could be fixed in a later PR. In the first incremental PR, we want to get identical results as current restart files.

Reply:
Agreed.